htlcswitch/hop/payload.go

```diff
@@ -74,12 +78,14 @@ func NewPayloadFromReader(r io.Reader) (*Payload, error) {
 		cid  uint64
 		amt  uint64
 		cltv uint32
+		mpp  = &record.MPP{}
```

Comment:
Seems this is mostly there to just pass in the proper record below? Alternatively, we can create a new function in the `record` package, just as we do for the current default routing types.

Reply:
not sure i follow, the struct is defined in this scope so that it can be set on the payload later (and nilled if we don't need it)

Reply:
Agree that it's a bit confusing. I don't know the underlying details of the TLV parsing, but what makes it confusing is that this empty `mpp` variable is used to create the `mpp.Record()` below.  Seems like we could just call a static `mpp.DeafaultRecord()` method, but I guess the variable is passed in by reference (as the other vars here) and modified by the underlying decoder?

Reply:
yeah doing something like `mpp.DefaultRecord()` means we never capture the decoded object and we'd just be throwing away the data. the `mpp.Record()` contains pointers to the fields in `mpp` so that they can be set by the decoder

is the difference just a matter of `NewMPPRecord(&mpp)` vs `mpp.Record()`? fwiw i went with the latter so that they can be easily unit tested, since there's now a single interface new records have to implement and we get testing for free.